src/logistration/LoginPage.jsx

```diff
@@ -147,6 +148,9 @@ class LoginPage extends React.Component {
     } = this.props;
     const { currentProvider, providers, secondaryProviders } = this.props.thirdPartyAuthContext;
 
+    const params = (new URL(window.location.href)).searchParams;
+    const activationMsgType = params.get('next') ? params.get('next').split('activate_account=')[1] : null;
```

Comment:
Why we are checking for `next` query param here? Can't we simply get the `activate_account` query param like this?
```suggestion
    const activationMsgType = params.get('activate_account');
```

Reply:
No. Because when user redirects to dashboard and is not authenticated, the query param next is added by the @login_required decorator and this just gets appended to the url 
```
http://localhost:1999/login?next=/dashboard?activate_account=error
```
if it was added with an `&` we could have fetched it directly
```
http://localhost:1999/login?next=/dashboard&activate_account=error
```

Reply:
Got it, but I guess we should avoid that extra redirect. Check my comment on the other PR.